source/interact-data/modify-results.txt

```diff
@@ -0,0 +1,281 @@
+.. _mongoid-data-modify-results:
+
+====================
+Modify Query Results
+====================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: ruby framework, odm, crud, print results, code example
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to customize the way that {+odm+}
+returns results to you from queries. MongoDB allows you to perform the
+following actions to modify the way that results appear:
+
+- :ref:`mongoid-data-projection`
+
+- :ref:`mongoid-data-sort`
+
+- :ref:`mongoid-data-skip-limit`
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``Band`` model, which represents a
+band or musical group. The definition of the ``Band`` model might be
+different for each section to demonstrate different query
+functionalities. Some sections might also use the ``Manager`` model,
+which represents a person who manages a given band, or a ``Tour`` model, which
+represents live performances by a certain band or musical group.
+
+.. _mongoid-data-projection:
+
+Return Specified Fields
+-----------------------
+
+In MongoDB, the process of specifying fields to include or exclude from
+results is called *projection*. {+odm+} provides the following operators
+to project fields:
+
+- ``only()``: Specifies fields to include
+- ``without()``: Specifies fields to exclude
+
+Include Fields
+~~~~~~~~~~~~~~
+
+The ``only()`` method retrieves only the specified fields from the
+database.
+
+The following code returns only the ``name`` field from documents in
+which the value of the ``members`` field is ``4``:
+
+.. literalinclude:: /includes/interact-data/modify-results.rb
+   :start-after: start-only
+   :end-before: end-only
+   :language: ruby
+   :dedent:
+
+.. note:: _id Field
+
+   In MongoDB, the ``_id`` field is included in results even if you do
+   not explicitly include it.
+
+If you attempt to reference attributes that have not been loaded,
+{+odm+} raises a ``Mongoid::Errors::AttributeNotLoaded`` error.
+
+You can also use the ``only()`` method to include fields from embedded
+documents. 
+
+Consider that the ``Band`` model embeds multiple ``Tour`` objects. You can
+project fields from the ``Tour`` model such as ``year``, as shown in the
+following code:
+
+.. literalinclude:: /includes/interact-data/modify-results.rb
+   :start-after: start-only-embed
+   :end-before: end-only-embed
+   :language: ruby
+   :dedent:
+
+Then, you can access the embedded fields from the returned documents:
+
+.. code-block:: ruby
+   
+   # Returns the first Tour object from
+   # the first Band in the results
+   bands.first.tours.first
+
+You can pass fields of referenced associations to the ``only()`` method,
+but the projection is ignored when loading the embedded objects. {+odm+}
+loads all fields of the referenced associations.
```

Comment:
Q: might be my lack of ruby experience, but i'm still not sure i understand this paragraph. will ruby people get it?

Reply:
It means that even if you project only one field of an embedded document, the whole thing will load when you access it from the result. I can reword